_posts/2018-08-27-deploy-digitalocean-ansible.md

```diff
@@ -0,0 +1,195 @@
+
+![Ansible logo](/images/deploy-digitalocean-ansible/ansible.png)
+
+Ansible is an open source based product by Red Hat that is described as _"a radically simple IT automation engine that automates cloud provisioning, configuration management, application deployment, intra-service orchestration, and many other IT needs"_.  Put simply, it enables automated deployment of the full stack, enabling developers and engineers to describe their deployments through a series of yaml based tasks that are grouped into plays.  These plays are collectively found in playbooks which are invoked through the `ansible-playbook` command.
+
+The playbook used here contains two plays:
+
+* the first play is executed on the control machine.  This uses the `dopy` module to drive the DigitalOcean API into creating a droplet as defined by a set of variables. Once the droplet starts to respond on port 22 the second play can commence.
+
+* the second play executes its tasks on the newly create droplet.  The control machine connects over SSH and executes a number of tasks to ready the droplet for deployment of OpenFaaS.  The number of tasks will vary depending upon the chosen orchestrator.
+
+![Ansible Schematic](/images/deploy-digitalocean-ansible/schematic.jpg)
+
+Once the playbook completes the reader will have a fully functioning single node OpenFaaS instance.
+
+## Pre-Requisites
+Ordinarily, a little pre-work is required in order to configure the automation environment, and as with most automation, the effort required upfront is far outweighed by the benefits borne out throughout its lifetime.  For the purpose of this article the Ansible based pre-reqs have been taken care of through the creation of a [Docker image](https://hub.docker.com/r/rgee0/ansible-playbook/) which already has [Python](https://www.python.org/), [pip](https://pip.pypa.io/en/stable/installing/), [dopy](https://pypi.org/project/dopy) and [Ansible](http://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html) installed.  Therefore, the only pre-requisite here is Docker on the machine where the playbook will be run from.
+
+```
+$ curl -sSL get.docker.com | sh
+```
+
+## DigitalOcean account
+In order to deploy onto [DigitalOcean](https://www.digitalocean.com/) the reader must have an account and enable API access by generating an API key. 
+> A DigitalOcean account with $10 of credit can be  [created here](https://m.do.co/c/2962aa9e56a1) - $10 is enough to try out this post and much more besides.
+
+### Create and upload an SSH key
+
+Ansible is agentless and typically uses SSH to perform its remote tasks, so it is essential that the public side of the key being used on the control machine is entered into the DigitalOcean control panel.  The play which will instruct the DigitalOcean API to create a new droplet will designate that this key should be deployed onto the new droplet upon instantiation.
```

Comment:
Is the target audience new users to Ansible? I think it would be nice to reference the "control machine" as " control machine (your laptop)".  It will be a little more beginner friendly.

Reply:
I'll add that on the first reference